animal_crossing.py

```diff
@@ -0,0 +1,193 @@
+from datetime import datetime, time, timezone
+import re
+import sqlite3
+import dateutil
+import dateutil.tz
+
+import config
+
+if config.bot.acnh_db is not None:
+	db = sqlite3.connect(config.bot.acnh_db)
+	db.row_factory = sqlite3.Row
+
+	# enable foreign key constraints
+	with db:
+		db.execute('PRAGMA foreign_keys = ON')
+
+friend_code_regex = re.compile(r'SW-\d{4}-\d{4}-\d{4}')
+cached_row_id = 0
+
+def stalk_market(cmd):
+	if not cmd.args:
+		# TODO: usage string? some other default behavior?
+		return
+
+	split = cmd.args.split(' ', 1)
+	subcmd = split[0]
+	if subcmd == 'tz':
+		_stalk_set_timezone(cmd, split)
+	elif subcmd == 'sell':
+		_stalk_sales(cmd, split)
+
+def _stalk_sales(cmd, split):
+	if len(split) != 2:
+		_stalk_list_sale_prices(cmd)
+		return
+	user_id = cmd.sender['id']
+	current_time = datetime.now(timezone.utc)
+	value = int(split[1])
+	try:
+		with db:
+			db.execute('''
+			INSERT INTO sale_price VALUES (?, ?, ?)
+			''', (user_id, current_time, value))
+		cmd.reply('Sale price recorded!')
+	except sqlite3.IntegrityError:
+		cmd.reply('Could not add sale price. Have you registered a friend code?')
+
+def _stalk_list_sale_prices(cmd):
+	# TODO: is there a better way to do this without a gross global?
+	global cached_row_id
+	cur = db.execute('''
+	SELECT sale_price.rowid, *
```

Comment:
that's a pretty crazy usage of `*`. at least just `SELECT *`?

Reply:
if I did `SELECT *` it wasn't returning the rowid, I had to explicitly ask for it.

Reply:
ohhh right